src/rpc/blockchain.cpp

```diff
@@ -1342,16 +1332,14 @@ UniValue getblockchaininfo(const JSONRPCRequest& request)
     }
 
     const Consensus::Params& consensusParams = Params().GetConsensus();
-    UniValue softforks(UniValue::VARR);
-    UniValue bip9_softforks(UniValue::VOBJ);
-    softforks.push_back(SoftForkDesc("bip34", 2, tip, consensusParams));
-    softforks.push_back(SoftForkDesc("bip66", 3, tip, consensusParams));
-    softforks.push_back(SoftForkDesc("bip65", 4, tip, consensusParams));
-    for (int pos = Consensus::DEPLOYMENT_CSV; pos != Consensus::MAX_VERSION_BITS_DEPLOYMENTS; ++pos) {
-        BIP9SoftForkDescPushBack(bip9_softforks, consensusParams, static_cast<Consensus::DeploymentPos>(pos));
```

Comment:
Thanks for digging that out. I really don't understand why that PR was merged. The author's justification (https://github.com/bitcoin/bitcoin/pull/10874#issuecomment-321609838) was "I had only made the change since I needed to add stats about BIP 91 to one of my branches and the fact that the forks needed to be hard coded in to be displayed with getblockchaininfo bothered me." It doesn't make sense to me that lines of code were added to bitcoin core to support someone's fork of the project.

When there are only three soft fork deployments, listing them out explicitly is less lines of code and seems clearer to me.

Reply:
(also, at the end of this PR, there's only one BIP9 softfork left (TESTDUMMY), so having a for loop _really_ doesn't make sense)